### PR TITLE
Adding a parser for Gitleaks scanner

### PR DESCRIPTION
**Pull request by steeve85**
Gitleaks (https://github.com/zricethezav/gitleaks) is a fast and customizable scanner to find secrets using regex and entropy in git repositories.

Documentation PR can be find here https://github.com/DefectDojo/Documentation/pull/83

**Note: DefectDojo is now on Python3.5 and Django 2.2.x Please submit your pull requests to the 'dev' branch as the 'legacy-python2.7' branch is only for bug fixes. Any new features submitted to the legacy branch will be ignored and closed.**

When submitting a pull request, please make sure you have completed the following checklist:

- [x] Your code is flake8 compliant 
- [x] Your code is python 3.5 compliant (specific python >=3.6 syntax is currently not accepted)
- [x] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [x] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [x] Add applicable tests to the unit tests.
- [x] Add the proper label to categorize your PR 

Current accepted labels for PRs:
- Import Scans (for new scanners/importers)
- enhancement

### CODE REVIEW

File: dojo/fixtures/test_type.json [hunk 1]
```diff
@@ -563,6 +563,13 @@
...
-    }
+		},
+	{
+			"fields": {
+					"name": "Gitleaks Scan"
+			},
+			"model": "dojo.test_type",
+			"pk": 173
+	}
 ]
```
Comment (valentijnscholten) [inline dojo/fixtures/test_type.json:574]: indentation seems to be off

**Comment by valentijnscholten** [comment]
Thanks, let's see if we can get Travis to build it succesfully as travis seems to be off lately.

File: dojo/tools/gitleaks/parser.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                                  test=test,
+                                  cwe=798,
+                                  active=False,
+                                  verified=False,
+                                  description=description,
+                                  severity=severity,
+                                  numerical_severity=Finding.get_numerical_severity(severity),
+                                  mitigation="Secrets and passwords should be stored in a secure vault and/or secure storage.",
+                                  impact="This weakness can lead to the exposure of resources or functionality to unintended actors, possibly providing attackers with sensitive information or even execute arbitrary code.",
+                                  references="N/A",
```
Comment (madchap) [inline dojo/tools/gitleaks/parser.py:54]: No need to add anything static there. That field is not required.

File: dojo/tools/gitleaks/parser.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                                  file_path=file_path,
+                                  url='N/A',
```
Comment (madchap) [inline dojo/tools/gitleaks/parser.py:56]: Same here. You can just remove that line.

File: dojo/fixtures/test_type.json (hunk 1 above)
Comment (steeve85) [reply dojo/fixtures/test_type.json:574]: I fixed the indentation in the new commit. Thanks.

File: dojo/tools/gitleaks/parser.py (hunk 2 above)
Comment (steeve85) [reply dojo/tools/gitleaks/parser.py]: This has been removed in the last commit. Thanks

File: dojo/tools/gitleaks/parser.py (hunk 3 above)
Comment (steeve85) [reply dojo/tools/gitleaks/parser.py]: This has been removed in the last commit. Thanks

**Comment by valentijnscholten** [comment]
I resolved conflicts, let's see if travis can build it and we can approve it.

**Comment by madchap** [comment]
Could you please submit a PR in https://github.com/DefectDojo/sample-scan-files as well? Thank you :)

**Comment by steeve85** [comment]
I just submitted a PR https://github.com/DefectDojo/sample-scan-files/pull/48

**Comment by valentijnscholten** [comment]
> I just submitted a PR [DefectDojo/sample-scan-files#48](https://github.com/DefectDojo/sample-scan-files/pull/48)

merged, thanks.

**Comment by 0xfabiof** [comment]
Hi, I haven't been able to submit a Gitleaks report via the API on 1.6rc version - I always get a bad-request error. Any ideas?

**Comment by valentijnscholten** [comment]
Open an issue, provide report + logs.

**Pull request merged**